### PR TITLE
Add basic codemirror modal

### PR DESCRIPTION
**Pull request by hardyjosh**
## Solution

image

## Checks

By submitting this for review, I'm confirming I've done the following:
- [x] made this PR as small as possible
- [x] unit-tested any new functionality
- [x] linked any relevant issues or PRs

### CODE REVIEW

**Comment by hardyjosh** [comment]
fixes #11

**Comment by hardingjam** [comment]
Fixed the CodeMirror to have a fixed height and be scrollable - looks ready to go

**Review by thedavidmeister — CHANGES_REQUESTED** [review]
no screenshots or tests

**Review by thedavidmeister — CHANGES_REQUESTED** [review]
judging by the screenshots, the bindings are not working (io ratio bindings are 0)

File: app/_components/CodemirrorModal.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+interface CodemirrorModalProps {
+	yamlData: YamlData;
+	currentState: FrameState;
+	dotrainText: string;
+	setError: React.Dispatch<SetStateAction<string | React.ReactElement | null>>;
+}
+
+export const CodemirrorModal = ({ currentState, dotrainText, yamlData }: CodemirrorModalProps) => {
+	console.log(yamlData);
```
Comment (thedavidmeister) [inline app/_components/CodemirrorModal.tsx:21]: console log

File: app/_components/CodemirrorModal.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	const { scenario } = getOrderDetailsGivenDeployment(
+		yamlData,
+		currentState.deploymentOption?.deployment || ''
+	);
+	console.log('scen', scenario);
```
Comment (thedavidmeister) [inline app/_components/CodemirrorModal.tsx:26]: console log

File: app/_components/CodemirrorModal.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	const [composedDotrainText, setComposedDotrainText] = useState<string>('');
+
+	const getComposedDotrainText = async () => {
+		console.log('currentState', currentState.bindings);
```
Comment (thedavidmeister) [inline app/_components/CodemirrorModal.tsx:30]: console log

**Review by thedavidmeister — CHANGES_REQUESTED** [review]

**Review by thedavidmeister — APPROVED** [review]

**Pull request merged**